acme/acme/crypto_util.py

```diff
@@ -0,0 +1,146 @@
+"""Crypto utilities."""
+import contextlib
+import logging
+import socket
+import sys
+
+from six.moves import range  # pylint: disable=import-error,redefined-builtin
+
+import OpenSSL
+
+from acme import errors
+
+
+logger = logging.getLogger(__name__)
+
+
+_DEFAULT_SSL_METHOD = OpenSSL.SSL.SSLv23_METHOD
+
+
+def _serve_sni(certs, sock, reuseaddr=True, method=_DEFAULT_SSL_METHOD,
+               accept=None):
```

Comment:
1. `acme` is meant for both client and server implementations
2. This was the only sensible way to unittest `_probe_sni`
3. I'm planning to refactor standalone (including the use of `_serve_sni`) in a separate PR.